src/janitor.c

```diff
@@ -366,6 +366,9 @@ static void pool_client_maint(PgPool *pool)
 	PgSocket *client;
 	usec_t age;
 
+	/* find and disconnect clients if the need to be closed */
+	for_each_client(pool, disconnect_client_if_close_needed);
```

Comment:
Although this should only need to run in the release_server method, I think it might make sense to run this as part of the janitor. I initially only had logic here but realized I need to be checking this as soon as a transaction is finished. Thoughts?